subsys/net/ip/tcp.c

```diff
@@ -1289,13 +1289,16 @@ int net_tcp_parse_opts(struct net_pkt *pkt, int opt_totlen,
 		}
 
 		switch (opt) {
-		case NET_TCP_MSS_OPT:
+		case NET_TCP_MSS_OPT: {
+			u16_t mss;
 			if (optlen != 2) {
 				goto error;
 			}
 			frag = net_frag_read(frag, pos, &pos,
```

Comment:
You could use net_frag_read_be16() here directly.

Reply:
Indeed, fixed.